plugins/by-name/nvim-tree/deprecations.nix

```diff
@@ -0,0 +1,182 @@
+lib: {
+  imports = [
+    (lib.mkRemovedOptionModule [
+      "plugins"
+      "nvim-tree"
+      "view"
+      "hideRootFolder"
+    ] "Set `plugins.nvim-tree.renderer.rootFolderLabel` to `false` to hide the root folder.")
+
+    # TODO: added 2025-04-07, remove after 25.05
+    (lib.nixvim.mkRemovedPackageOptionModule {
+      plugin = "nvim-tree";
+      packageName = "git";
+    })
+
+  ];
+
+  # Deprecations after mkNeovimPlugin migration
+  deprecateExtraOptions = true;
+
+  optionsRenamedToSettings =
+    let
+      mkOptionPaths = map (lib.splitString ".");
```

Comment:
Clever way to improve how the option paths are formatted!

Personally, I'd inline it to remove the let..in block, though:

```nix
optionsRenamedToSettings = map (lib.splitString ".") [
  "disableNetrw"
  "hijackNetrw"
  "autoReloadOnWrite"
  #...
```